### PR TITLE
Fix findminmax! for OffsetArray

### LINKED ISSUE #83: Wrong results using argmin or argmax
Using `argmin` or `argmax` along a dimension of an OffsetArray gives wrong results for some offsets
Examples:
```julia
using OffsetArrays
A = [1 2; 1 2]
for offset in 0:-1:-3
    A_off = OffsetArray(A, offset, -1)
    A_off[argmin(A_off, dims=2)] |> println
end
```
produces
```julia
[1; ; 1] # Correct
[2; ; 1] # Wrong
[1; ; 2] # Wrong
[1; ; 1] # Correct
```

### PR DESCRIPTION
**Pull request by ghost**
Fix JuliaArrays/OffsetArrays.jl#83

### CODE REVIEW

File: base/reducedim.jl [hunk 1]
```diff
@@ -695,7 +699,7 @@ function findminmax!(f, Rval, Rind, A::AbstractArray{T,N}) where {T,N}
     keep, Idefault = Broadcast.shapeindexer(indsRt)
     ks = keys(A)
     y = iterate(ks)
-    zi = zero(eltype(ks))
+    zi = _get_unused_index(A)
```
Comment (timholy) [inline base/reducedim.jl]: For safety, I think you want to make this `_get_unused_index(first(ks))` and accommodate the different types of keys. You can't promise that someone hasn't written a specialized `keys` function, and you'll note below that what gets stored in `Rind` is an entry in `ks`.

File: base/reducedim.jl [hunk 2]
```diff
@@ -742,7 +746,7 @@ dimensions of `rval` and `rind`, and store the results in `rval` and `rind`.
 """
 function findmin!(rval::AbstractArray, rind::AbstractArray, A::AbstractArray;
                   init::Bool=true)
-    findminmax!(isless, init && !isempty(A) ? fill!(rval, first(A)) : rval, fill!(rind,zero(eltype(keys(A)))), A)
+    findminmax!(isless, init && !isempty(A) ? fill!(rval, first(A)) : rval, fill!(rind, _get_unused_index(A)), A)
```
Comment (timholy) [inline base/reducedim.jl]: Here and everywhere else you want to again use `_get_unused_index(first(keys(A)))`.

**Review by timholy — CHANGES_REQUESTED** [review]
Thanks very much @mikan2087, this is very good work. I note there are test failures on several of the buildbots, and they look related to this PR. I've made some suggestions below that hopefully will fix the failures. Then this will be good to go.

Thanks for your contributions!

**Comment by timholy** [comment]
Just bumping this as it would be nice to get this fixed.

**Comment by DilumAluthge** [comment]
It looks like this pull request is no longer mergeable.

If you would still like to make this change, please make a new pull request based on the latest `master` branch of Julia.

**Pull request closed without merging**